disneyland/server.go

```diff
@@ -18,45 +18,42 @@ func detailedInternalError(err error) error {
 		fmt.Sprintf("Error processing job: %v", err),
 	)
 }
-func validateAccess(project string, userProject string, kind string, userKind string) (string, string, error) {
-	if kind == "" {
-		if userKind == "ANY" {
-			return "", "", grpc.Errorf(codes.DataLoss, "ListJobsKindRequest.Kind not specified")
-		}
-		kind = userKind
+func (s *Server) Init() {}
+func (u *User) ValidateUserAccess() bool {
+	if u.ProjectAccess != "ANY" && u.KindAccess == "ANY" {
+		return true
 	}
-	if project == "" {
-		if userProject == "ANY" {
-			return "", "", grpc.Errorf(codes.DataLoss, "ListJobsKindRequest.Project not specified")
-		}
-		project = userProject
+	return false
+}
+func (u *User) ValidateWorkerAccess() bool {
+	if u.ProjectAccess == "ANY" && u.KindAccess != "ANY" {
+		return true
 	}
+	return false
+}
 
-	if userKind != kind && userKind != "ANY" {
-		return "", "", grpc.Errorf(codes.PermissionDenied, "job.Kind ≠ user.KindAccess")
+func (u *User) ValidateJobAccess(in *Job) bool {
+	// if worker
+	if u.ValidateWorkerAccess() && in.Kind != u.KindAccess {
+		return false
 	}
-	if userProject != project && userProject != "ANY" {
-		return "", "", grpc.Errorf(codes.PermissionDenied, "job.Project ≠ user.ProjectAccess")
+	// if user
+	if u.ValidateUserAccess() && in.Project != u.ProjectAccess {
+		return false
 	}
-	return project, kind, nil
-}
-func (s *Server) Init() {
+	return true
 }
 
 func (s *Server) CreateJob(ctx context.Context, in *Job) (*Job, error) {
 	user := getAuthUserFromContext(ctx)
-	userProject := user.ProjectAccess
-	project := in.Project
-	if project == "" {
-		if userProject == "ANY" {
-			return nil, grpc.Errorf(codes.DataLoss, "Job.Project not specified")
-		}
-		project = userProject
-	}
-	if userProject != project && userProject != "ANY" {
-		return nil, grpc.Errorf(codes.PermissionDenied, "job.Project ≠ user.ProjectAccess")
-	}
-	in.Project = project
+
+	// if worker
+	if user.ValidateWorkerAccess() {
```

Comment:
I guess `ValidateWorkerAccess` name does not make sense in this context.  `if user.IsWorker() {` might work much better